### PR TITLE
[libc++abi] Fix test on Android

### PR DESCRIPTION
**Pull request by maksfb**
Follow up to #72727. The added test could not be executed on Android.

### CODE REVIEW

**Comment by maksfb** [comment]
Checked that all test failures are unrelated.

**Comment by ldionne** [comment]
@rprichard Can you take a look? I suspect the right thing to do here is not to mark the test as `UNSUPPORTED`, since that would simply reduce your test coverage.

**Comment by rprichard** [comment]
I think the `UNSUPPORTED` is correct here.

The `RUN: %t.exe` line should be `RUN: %{exec} %t.exe`, but after fixing that, the test will still fail on Android because of the `-no-pie` compile option in `RUN: %{cxx} %{flags} %s %{link_flags} -no-pie -o %t.exe`. The Android dynamic loader rejects non-PIE executables:
```
"./t.tmp.exe": error: Android 5.0 and later only support position-independent executables (-fPIE).
```

I can't make the executable work as a PIE because the address of the landing pad would require a dynamic relocation, which can't be applied because `.gcc_except_table` is read-only. Making the section writable instead is rejected by the assembler:
```
/llvm/libcxxabi/test/native/x86_64/lpstart-zero.pass.sh.s:76:2: error: changed section flags for .gcc_except_table, expected: 0x2
        .section        .gcc_except_table,"aw",@progbits
        ^
```

Also:

```
.Lexception0:
	.byte	3                               # @LPStart Encoding = udata4
  .long 0
	.byte	155                             # @TType Encoding = indirect pcrel sdata4
	.uleb128 .Lttbase0-.Lttbaseref0
.Lttbaseref0:
	.byte	11                              # Call site Encoding = udata4
```

`DW_EH_PE_udata4` is 3 but `DW_EH_PE_sdata4` is 11. I think the comment on the call side encoding is wrong -- it should be `sdata4` instead of `udata4`.

File: libcxxabi/test/native/x86_64/lpstart-zero.pass.sh.s [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # RUN: %t.exe
 
 # REQUIRES: linux && target={{x86_64-.+}}
-# UNSUPPORTED: no-exceptions
+# UNSUPPORTED: no-exceptions || android
```
Comment (ldionne) [inline libcxxabi/test/native/x86_64/lpstart-zero.pass.sh.s:5]: @rprichard  What would be the right target-based annotation here?

```
// UNSUPPORTED: target={{.+}}-android{{.+}}
```

?

**Review by ldionne — COMMENTED** [review]
Thanks for the analysis @rprichard!

@maksfb Can you please address those comments?

**Comment by maksfb** [comment]
Fixed the execution command and the comment. The test is supposed to be non-PIE, which excludes it from testing on Android.

File: libcxxabi/test/native/x86_64/lpstart-zero.pass.sh.s (hunk 1 above)
Comment (rprichard) [reply libcxxabi/test/native/x86_64/lpstart-zero.pass.sh.s:5]: I think I would change the suffix to use `{{.*}}` to also match a triple without the version number suffix, just for completeness:

```
// UNSUPPORTED: target={{.+}}-android{{.*}}
```

The two unversioned Android annotations I currently see in libc++ also use a `{{.*}}` suffix:

```
// XFAIL: LIBCXX-ANDROID-FIXME && target={{i686|x86_64}}-{{.+}}-android{{.*}}
```
```
// UNSUPPORTED: target=i686-{{.+}}-android{{.*}}
```

**Review by ldionne — APPROVED** [review]

**Pull request merged**